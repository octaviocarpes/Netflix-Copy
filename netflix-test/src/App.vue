<template>
  <div id="app">
    <Loader />
    <router-view />
  </div>
</template>

<script>
import store from "@/store";
import Loader from "@/components/loader/Loader";
import StorageService from "@/storage";

export default {
  components: {
    Loader
  },

  mounted() {
    if (!StorageService.getStorage()) StorageService.initiateStorage();

    const user = StorageService.getUserSession();
    if (user) store.dispatch("PERFORM_LOGIN", user);
  }
};
</script>

<style lang="scss">
* {
  padding: 0px;
  margin: 0px;
}

html,
body {
  position: absolute;
  background-color: black;
  width: 100%;
  height: 100%;
  overflow-x: hidden;
}

#app {
  font-family: "Avenir", Helvetica, Arial, sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  text-align: center;
  color: #2c3e50;
  position: relative;
  height: 100%;
  width: 100%;
}
#nav {
  padding: 30px;
  a {
    font-weight: bold;
    color: #2c3e50;
    &.router-link-exact-active {
      color: #42b983;
    }
  }
}
</style>
