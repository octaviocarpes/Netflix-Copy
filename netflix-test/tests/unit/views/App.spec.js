import Vue from "vue";
import App from "@/App";
import { shallowMount } from '@vue/test-utils'

describe("App.spec.js", () => {
  let cmp, vm;

  beforeEach(() => {
    shallowMount(App, {
      stubs: ['router-link', 'router-view']
    })
    cmp = Vue.extend(App);
    vm = new cmp({}).$mount();
  });

  it("has the expected html structure", () => {
    expect(vm.$el).toMatchSnapshot();
  });
});