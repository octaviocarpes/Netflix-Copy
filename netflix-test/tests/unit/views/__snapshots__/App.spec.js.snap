// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.spec.js has the expected html structure 1`] = `
<div
  id="app"
>
  <!---->
   
  <router-view />
</div>
`;
